pkg/cmd/marathon/app/app_list.go

```diff
@@ -1,25 +1,211 @@
 package app
 
 import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
 	"github.com/dcos/dcos-cli/api"
-	"github.com/dcos/dcos-core-cli/pkg/cmd/marathon/python"
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/dcos/dcos-core-cli/pkg/marathon"
+	"github.com/dustin/go-humanize"
+	m "github.com/gambol99/go-marathon"
+	"github.com/olekukonko/tablewriter"
 	"github.com/spf13/cobra"
 )
 
+const defaultTableValue = "---"
+
+// `truncate` will break if this is less than 3
+const maxTableValueLength = 30
+
+var deploymentDisplay = map[string]string{
+	"ResolveArtifacts":   "artifacts",
+	"ScaleApplication":   "scale",
+	"StartApplication":   "start",
+	"StopApplication":    "stop",
+	"RestartApplication": "restart",
+	"ScalePod":           "scale",
+	"StopPod":            "stop",
+	"RestartPod":         "restart",
+	"KillAllOldTasksOf":  "kill-tasks",
+}
+
 func newCmdMarathonAppList(ctx api.Context) *cobra.Command {
-	var json bool
+	var jsonOutput bool
 	var quiet bool
 
 	cmd := &cobra.Command{
 		Use:   "list",
 		Short: "List the installed applications.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return python.InvokePythonCLI(ctx)
+			client, err := marathon.NewClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			applications, err := client.Applications()
+			if err != nil {
+				return err
+			}
+
+			if quiet {
+				for _, app := range applications {
+					fmt.Fprintln(ctx.Out(), app.ID)
+				}
+				return nil
+			}
+
+			if jsonOutput {
+				enc := json.NewEncoder(ctx.Out())
+				enc.SetIndent("", "    ")
+
+				return enc.Encode(&applications)
+			}
+
+			deployments, err := client.Deployments()
+			if err != nil {
+				return err
+			}
+
+			deploymentMap := make(map[string]m.Deployment)
+			for _, d := range deployments {
+				deploymentMap[d.ID] = d
+			}
+
+			queue, err := client.Queue()
+			if err != nil {
+				return err
+			}
+
+			tableHeader := []string{"ID", "MEM", "CPUS", "TASKS", "HEALTH", "DEPLOYMENT", "WAITING", "CONTAINER", "CMD", "ROLE"}
+			table := cli.NewTable(ctx.Out(), tableHeader)
+			table.SetAlignment(tablewriter.ALIGN_LEFT)
+
+			items := make([][]string, 0, len(applications))
+			for _, app := range applications {
+				item := []string{
+					app.ID,
+					humanize.Ftoa(*app.Mem),
+					humanize.Ftoa(app.CPUs),
+					formatTaskRunning(app),
+					formatHealth(app),
+					formatDeployments(app, deploymentMap),
+					formatWaiting(app, queue),
+					formatContainerType(app),
+					formatCmd(app),
+					formatRole(app),
+				}
+				items = append(items, item)
+			}
+			sort.Sort(appRows(items))
+			table.AppendBulk(items)
+			table.Render()
+
+			return nil
 		},
 	}
 
-	cmd.Flags().BoolVar(&json, "json", false, "Print JSON-formatted data.")
+	cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print JSON-formatted data.")
 	cmd.Flags().BoolVarP(&quiet, "quiet", "q", false, "Display IDs only.")
 
 	return cmd
 }
+
+func truncate(s string) string {
```

Comment:
To make this func declaration better comprehensible I suggest this:
```suggestion
func truncate(s string, maxLength int) string {
```